packages/config/src/ProfileInfo.ts

```diff
@@ -179,6 +181,116 @@ export class ProfileInfo {
+                            loadedConfig.baseProfile = { type: "base", schema: value };
+                        } else if (key.indexOf(":base") < 0 && !loadedConfig.profiles.find(p => p.type === key.split(":")[1])) {
+                            // Add the schema corresponding to the given profile type
+                            loadedConfig.profiles.push({ type: key.split(":")[1], schema: value });
+                        }
+                    });
+                    ImperativeConfig.instance.loadedConfig = loadedConfig;
+                }
+
+                await ConfigAutoStore._storeSessCfgProps({
+                    defaultBaseProfileName: this.mLoadedConfig?.properties.defaults.base,
+                    sessCfg: {
+                        [options.property === "host" ? "hostname" : options.property]: options.value
+                    },
+                    propsToStore: [options.property],
+                    profileName: options.profileName,
+                    profileType: options.profileType,
+                    setSecure: options.setSecure
+                });
+            } else {
+                const profMgr = new CliProfileManager({ profileRootDirectory: this.mOldSchoolProfileRootDir, type: options.profileType });
+                // Add new property
+                await profMgr.update({ name: options.profileName, merge: true, profile: { [options.property]: options.value } });
+
+                // Update mOldSchoolProfileCache to get mergedArgs updated
+                this.mOldSchoolProfileCache.find(v => v.name === options.profileName).profile[options.property] = options.value;
+            }
+        }
+    }
+
+    /**
+     * Update a given property with the value provided.
+     * This function only works for properties that can be found in the config files (including secure arrays).
+     * If the property cannot be found, this function will resolve to false
+     * This function supports v1 profiles
+     * @param options Set of options required to update a known property
+     */
+    public async updateKnownProperty(options: IProfInfoUpdateKnownPropOpts): Promise<boolean> {
+        const toUpdate = options.mergedArgs.knownArgs.find((v => v.argName === options.property)) ||
+            options.mergedArgs.missingArgs.find((v => v.argName === options.property));
+
+        if (toUpdate == null) {
+            return false;
+        }
+
+        switch (toUpdate.argLoc.locType) {
+            case ProfLocType.OLD_PROFILE: {
+                const filePath = toUpdate.argLoc.osLoc;
+                const profileName = ProfileIO.fileToProfileName(filePath[0], "." + filePath[0].split(".").slice(-1)[0]);
+                const profileType = filePath[0].substring(this.mOldSchoolProfileRootDir.length + 1).split("/")[0];
+                const profMgr = new CliProfileManager({ profileRootDirectory: this.mOldSchoolProfileRootDir, type: profileType });
+                if (options.value != null) {
```

Comment:
Wondering if we ought to allow folks to set properties to `null` in a profile. I believe I have seen this done before in v1 profiles. Should we check if `options.value !== undefined` here instead?